### PR TITLE
Reorgnize simplerunner in to more testable pieces

### PR DESCRIPTION
**Pull request by fsfod**
Refactored simplerunner to be more like a module to help make testing it easier and to help with running benchmarks in a child process where the code flow is much different from in process.

All functions should now be local or moved to a new runner lib table to avoid polluting the global table that could of conflict with any benchmark that might of used the same named global. 
The initialization of various modules and systems is now all done in a single function instead of being spread around the whole file where they would get instantly run when the file is loaded

### CODE REVIEW

File: simplerunner.lua [hunk 1]
```diff
@@ -57,35 +57,33 @@ function runner.init()
...
-            runner.jitfunc(startimer)
-            runner.jitfunc(stoptimer)
-        else
-            startimer, stoptimer = runner.maketimer(os.clock)
-            -- os.clock can't be jit'ed so try to keep timer a bit more stable
-            jit.off(startimer)
-            jit.off(stoptimer)
-        end
+    local timeloaded, timelib = pcall(require, "time")
+    -- jit.util is need so we can check the bytecode has changed for the timer functions being pre JIT'ed.
```
Comment (ltratt) [inline simplerunner.lua]: "is needED"

**Comment by ltratt** [comment]
I only have one minor comment as things stand. There are a couple of grammar issues in commit messages, but will they be squashed away eventually?

**Comment by fsfod** [comment]
i would guess so if there not the first commit

**Comment by ltratt** [comment]
OK, so in the first commit: "should of" -> "should have"; "might of" -> "might have"; sentence after "and get instantly run" needs rephrasing. You can squash (make sure you fix my other comment too please!) at the same time.

**Comment by ltratt** [comment]
@fsfod What is the status of this?

**Comment by fsfod** [comment]
@ltratt I did try to improve the messages a bit

**Comment by fsfod** [comment]
But i did find 2 small bugs that you can see commits for

**Comment by ltratt** [comment]
OK, the new commit and the messages look reasonable. Can you squash and wrap the commit messages to 80 chars at the same time please?

**Comment by fsfod** [comment]
@ltratt squashed and ready to merge

**Pull request merged**